### PR TITLE
chore: add new fields to trial logs [DET-4176]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change just adds the new fields that will be used for filtering logs to the database and migrations to create them. Additionally, a new field is added for new log messages written by Fluentd so they can be reconstructed to match the previous trial logs.

## Test plan
- [x] read<->write fully saturated log with postgres backend

## Commentary (optional)
Not really much going on here, @dzhu if you wanted to last this with "write logs to the master through fluentd" instead, I'd be fine with that. But for now, I will use this to work on filtering with the new api.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: master/pkg/model/experiment.go [hunk 1]
```diff
@@ -390,6 +390,15 @@ type TrialLog struct {
...
 	Message string `db:"message"`
+
+	AgentID string `db:"agent_id"`
+	// In the case of k8s, container_id is a pod name instead.
+	ContainerID string    `db:"container_id"`
+	RankID      int       `db:"rank_id"`
+	Timestamp   time.Time `db:"timestamp"`
+	Level       string    `db:"level"`
+	StdType     int       `db:"std_type"`
+	Source      string    `db:"source"`
```
Comment (stoksc) [inline master/pkg/model/experiment.go:401]: Open to a better name since `StdType` itself is actually called `Source` in fluentd.

Comment (vishnu2kmohan) [reply master/pkg/model/experiment.go:401]: `StreamType`?

Comment (stoksc) [reply master/pkg/model/experiment.go:401]: Yeah, I had just called it `Stream` originally. But it's `StdType` in docker internally and `Source` in fluentd so I just went with `StdType`. `StreamType` maybe is more intuitive.

**Comment by stoksc** [comment]
@dzhu do you mind approving this? I approve of the change you made, but can't approve my own PR, so it's good on my end.

**Review by dzhu — APPROVED** [review]
Could you finish filling out the description (and make sure the full thing gets into the squashed commit message)?

**Pull request merged**